tests/unit/test_guarded_ai.py

```
@@ -0,0 +1,364 @@
+        call_args = mock_generate_feedback.call_args[0]
+        self.assertEqual(call_args[0], category)  # category
+        self.assertEqual(call_args[1], question)  # question
+        self.assertEqual(call_args[2], user_response)  # user_response
+
+        # Check tokens_for_ai includes language and transition instructions
+        tokens_arg = call_args[3]
+        self.assertIn("English", tokens_arg)
+        self.assertIn("Additional transition-specific instructions", tokens_arg)
+
+        # Check metadata was filtered
+        filtered_metadata = call_args[4]
```

Ruff (0.12.2)

Local variable `filtered_metadata` is assigned to but never used

Remove assignment to unused variable `filtered_metadata`

(F841)